app/models/merchant.rb

```diff
@@ -11,8 +11,23 @@ def total_revenue
       .where(transactions: { result: 'success' })
       .joins(:invoice_items)
       .sum('invoice_items.quantity * invoice_items.unit_price')
-    #start with invoices and transactions
-    #transaction need to have result success to be paid invoice
-    #total_revnue = quantity * unit_price (only invoice item table has relationship)
+  end
+
+  def self.most_revenue(quantity = nil)
+    select('merchants.*, SUM(invoice_items.quantity * invoice_items.unit_price) AS revenue')
+      .joins(invoices: [:transactions, :invoice_items])
+      .merge(Transaction.successful)
```

Comment:
Transaction.successful works due to 
`scope :successful, -> { where(result: "success") }` 
in Transaction model